src/valkey/js_valkey.zig

```diff
@@ -1073,14 +1107,40 @@ pub const JSValkeyClient = struct {
         return memory_cost;
     }
 
+    fn deinitSocketContextNextTick(this: *JSValkeyClient) void {
+        if (this._socket_ctx) |ctx| {
+            this._socket_ctx = null;
+            // socket close can potentially call JS so we need to enqueue the deinit
+            // this should only be the case tls socket with custom config
+            const Holder = struct {
+                ctx: *uws.SocketContext,
+                task: jsc.AnyTask,
+
+                pub fn run(self: *@This()) void {
+                    defer bun.default_allocator.destroy(self);
+                    self.ctx.deinit(true);
+                }
+            };
+            var holder = bun.handleOom(bun.default_allocator.create(Holder));
+            holder.* = .{
+                .ctx = ctx,
+                .task = undefined,
+            };
+            holder.task = jsc.AnyTask.New(Holder, Holder.run).init(holder);
+
+            this.client.vm.enqueueTask(jsc.Task.init(&holder.task));
+        }
```

Comment:
```suggestion
        const ctx = this._socket_ctx orelse return;

        this._socket_ctx = null;
        // socket close can potentially call JS so we need to enqueue the deinit
        // this should only be the case tls socket with custom config
        const Holder = struct {
            ctx: *uws.SocketContext,
            task: jsc.AnyTask,

            pub fn run(self: *@This()) void {
                defer bun.default_allocator.destroy(self);
                self.ctx.deinit(true);
            }
        };
        var holder = bun.handleOom(bun.default_allocator.create(Holder));
        holder.* = .{
            .ctx = ctx,
            .task = undefined,
        };
        holder.task = jsc.AnyTask.New(Holder, Holder.run).init(holder);

        this.client.vm.enqueueTask(jsc.Task.init(&holder.task));
```

Easier to follow when you exit early.